.github/workflows/publish-docker.yml

```diff
@@ -0,0 +1,97 @@
+name: Build and Publish Docker
+
+on:
+    push:
+      tags:
+        - "v*.*.*"
+    schedule:
+      - cron: "0 0 * * *"
+    # Trigger without any parameters a proactive rebuild
+    workflow_dispatch: {}
+    workflow_call:
+
+env:
```

Comment:
We use docker hub as a registry, and the location is actually a16zcrypto/magi rather than a16z/magi